inject-generator/src/main/java/io/avaje/inject/generator/ProcessingContext.java

```diff
@@ -188,7 +198,28 @@ static void processingOver(boolean over) {
   }
 
   static void writeSPIServicesFile() {
+
+    try (final var file =
+            APContext.filer()
+                .getResource(StandardLocation.CLASS_OUTPUT, "", Constants.META_INF_SPI)
+                .toUri()
+                .toURL()
+                .openStream();
+        final var buffer = new BufferedReader(new InputStreamReader(file)); ) {
+
+      String line;
+      while ((line = buffer.readLine()) != null) {
+        line.replaceAll("\\s", "")
+            .replace(",", "\n")
+            .lines()
+            .forEach(ProcessingContext::addInjectSPI);
```

Comment:
We need to read and register these because at line 225 that is a write **_replace_** and not a write _**append**_ ?

Reply:
yes